alerting/src/main/kotlin/org/opensearch/alerting/MonitorRunnerService.kt

```diff
@@ -218,32 +224,31 @@ object MonitorRunnerService : JobRunner, CoroutineScope, AbstractLifecycleCompon
     }
 
     override fun runJob(job: ScheduledJob, periodStart: Instant, periodEnd: Instant) {
-        if (job !is Monitor) {
-            throw IllegalArgumentException("Invalid job type")
-        }
-        launch {
-            runJob(job, periodStart, periodEnd, false)
+        when (job) {
+            is Workflow -> {
+                launch {
+                    runJob(job, periodStart, periodEnd, false)
+                }
+            }
+            is Monitor -> {
+                launch {
+                    runJob(job, periodStart, periodEnd, false)
+                }
+            }
+            else -> {
+                throw IllegalArgumentException("Invalid job type")
+            }
         }
     }
 
-    suspend fun runJob(job: ScheduledJob, periodStart: Instant, periodEnd: Instant, dryrun: Boolean): MonitorRunResult<*> {
+    suspend fun runJob(workflow: Workflow, periodStart: Instant, periodEnd: Instant, dryrun: Boolean): WorkflowRunResult {
+        return CompositeWorkflowRunner.runWorkflow(workflow, monitorCtx, periodStart, periodEnd, dryrun)
+    }
 
-        // Updating the scheduled job index at the start of monitor execution runs for when there is an upgrade the the schema mapping
-        // has not been updated.
-        if (!IndexUtils.scheduledJobIndexUpdated && monitorCtx.clusterService != null && monitorCtx.client != null) {
```

Comment:
We need this to make sure the config index is updated before monitors/workflows execute to prevent mapping problems.

Reply:
reverted. thanks.